src/pages/library/DataBrowserPreview.js

```diff
@@ -42,6 +42,38 @@ const styles = {
 
 const activeTab = 'datasets'
 
+const DatasetPreviewSelector = ({ access, dataset, selectedTable, setSelectedTable, selectOptions }) => Utils.switchCase(access,
```

Comment:
This abstraction seemed like a nice driveby change. Also cause it wasn't really behaving well as an inline switchCase